### PR TITLE
Make "cm" module compatible with Debian Jessie

### PR DESCRIPTION
**Pull request by ppp0**
part of https://github.com/cargomedia/puppet-cargomedia/issues/1157

### CODE REVIEW

**Comment by ppp0** [comment]
Some file have been erroneously committed that will disappear when merging upstream post-#1371-merge

**Comment by ppp0** [comment]
@kris-lab not much left to review :) - still, please review

**Comment by kris-lab** [comment]
lgtm, let's wait for build.

**Comment by ppp0** [comment]
retest this please

**Comment by kris-lab** [comment]
Looks like it's falling still?

**Comment by ppp0** [comment]
I interrupted it :( but it is failing too, on a mongo timeout

**Comment by ppp0** [comment]
A spec started to fail apparently due to a security update to libssl. Error was thrown by `curl` trying to connect to `socat` (https://github.com/cargomedia/puppet-packages/blob/master/modules/cm/spec/reverse-proxy/real-ip-header/manifest.pp#L10 and https://github.com/cargomedia/puppet-packages/blob/master/modules/cm/spec/reverse-proxy/real-ip-header/spec.rb#L6)  yielding
````
SSL3_CHECK_CERT_AND_ALGORITHM:dh key too small
```

The solution was to add a big enough dh number to the cert file (as outlined in https://youremindmeofmymother.com/2015/08/21/socat-ssl-ssl-routinesssl3_check_cert_and_algorithmdh-key-too-small/)

This might well affect other spec tests in our suite once migrated to Jessie (because of the newer libssl library there)

cc @kris-lab @njam

**Comment by ppp0** [comment]
@njam @kris-lab please review (cross fingers that specs finally will succeed)

**Comment by ppp0** [comment]
````
Finished in 2 hrs 5 mins 33 secs.
Success! 72 specs run, 0 failures (352 examples, 0 failures)
````

**Comment by ppp0** [comment]
retest this please

**Comment by kris-lab** [comment]
As far as I see the sieze of DH (Diffie-Hellman) has being increased for many times over time.

Why does it affect us now and not in the past?

**Comment by ppp0** [comment]
To be honest I don't know. To me, it looks like this is being enforced in libssl or a similar shared library external to both client and server. The error on the server (socat) looks different: it says the SSL handshake could not be successfully completed.

I believe this enforcement of minimum DH key size is a newer thing. Plus, I believe we'll see this problem in other places too in the future. Otoh it's good to know there is a trivial solution to this.

PR mergeable?

File: modules/cm/templates/spec/spec-ssl.pem [hunk 1]
```diff
@@ -16,3 +16,8 @@ eJVbA3znNP4LbxylcY9MZGMDh066/hR53Br8VSJ6KIpBPk3Aki+DtS17HWpxx9AM
 2i5UY00P3wUo0LsAOscQ+MTRchSD74G2P55FVkrejkkxTKn5iMT3d+2MTaM7Dybg
 /0Ms/44HxA/0j3dwSxLLtGc8Vk6Gye3TeLd0z5volImOHg==
 -----END CERTIFICATE-----
+-----BEGIN DH PARAMETERS-----
+MIGHAoGBAL//R/T0xbr7f+sn8iqW1eULbb9rq4cQCNSMe4y7FdYX0uFqgPcyFtxy
+DmsTjijoyshFBR2ipOPrfWG5G5hT7jVu/eCn+544EUQ5YToSAYqFhlchYp1s4AE2
+vlr1bVNG1hqSertCj/YdbgF/qmKU5tJcdGanwca8Bm2mu1yEy7LrAgEC
+-----END DH PARAMETERS-----
```
Comment (ppp0) [inline modules/cm/templates/spec/spec-ssl.pem]: This was generated by issuing
```
openssl dhparam -out dhparams.pem 1024; cat dhparams.pem >> spec-ssl.pem
```

**Comment by kris-lab** [comment]
PR mergeable imo, once the build is done.

**Comment by kris-lab** [comment]
About DH.... agree, from what I read it seems we will be forced to introduce DH on some point. It simply protects agains some hacking scenarios.

**Pull request merged**